File: src/Entity/Browser/Container/Page/Content/Plain.php
<?php

declare(strict_types=1);

namespace Yggverse\Yoda\Entity\Browser\Container\Page\Content;

use \GdkEvent;
use \GtkLabel;

use \Yggverse\Yoda\Model\Gtk\Pango\Markup;

class Plain extends \Yggverse\Yoda\Abstract\Entity\Browser\Container\Page\Content\Markup
{
    public function set(
        string $source
    ): void
    {
        $this->gtk->set_markup(
            Markup::code( // @TODO
                $this->_source = $source
            )
        );
    }
}

File: src/browser/main/tab/page/content/text/gemini/reader/parser/plain.rs
use gtk::glib::{markup_escape_text, GString};

pub struct Plain {
    markup: GString,
    source: GString,
}

impl Plain {
    pub fn from(line: &str) -> Plain {
        Self {
            markup: GString::from(format!("{}\n", markup_escape_text(line))),
            source: GString::from(line),
        }
    }

    pub fn markup(&self) -> &GString {
        &self.markup
    }

    pub fn source(&self) -> &GString {
        &self.source
    }
}
